### PR TITLE
TRACING-3270| Document OpenTelemetry prometheus exporter

### PR DESCRIPTION
**Pull request by pavolloffay**
Version(s):
4.11, 4.12, 4.13, 4.14

Issue:

https://issues.redhat.com/browse/TRACING-3270

Link to docs preview:

https://64716--docspreview.netlify.app/

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by pavolloffay** [comment]
The PR can be merged ASAP, it's for RHOSDT 2.9

File: modules/distr-tracing-otel-config-collector.adoc [hunk 1]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+        enable_open_metrics: true <6>
+        resource_to_telemetry_conversion: <7>
+          enabled: true
+        metric_expiration: 180m <8>
+    service:
+      pipelines:
+        metrics:
+          exporters: [jaeger]
+----
+<1> Exposes Prometheus port from the collector Pod and Service. The port name can be then used in `ServiceMonitor` or `PodMonitor` to enable scraping of metrics by the Prometheus.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:432]:
```suggestion
<1> Exposes Prometheus port from the collector Pod and Service. You can enable scraping of metrics by the Prometheus by using the port name in `ServiceMonitor` or `PodMonitor`.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 2]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+<2> The endpoint on which the metrics will be exposed.
+<3> The server side TLS configuration. Defines paths to TLS certificates.
+<4> If set, exports metrics under the provided value. No default.
+<5> Label key/values that are applied for every exported metric. No default.
+<6> If true, metrics will be exported using the OpenMetrics format. Exemplars are only exported in the OpenMetrics format, and only for histogram and monotonic sum (i.e. counter) metrics. Disabled by default.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:437]:
```suggestion
<6> If true, metrics will be exported using the OpenMetrics format. Exemplars are only exported in the OpenMetrics format and only for histogram and monotonic sum metrics as such `counter`. Disabled by default.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 3]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+<7> If `enabled` is `true`, all the resource attributes will be converted to metric labels by default. Disabled by default.
+<8> Defines how long metrics are exposed without updates. Default is `5m`.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:439]:
```suggestion
<8> Defines how long metrics are exposed without updates; `5m` is the default.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 4]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+          enabled: true
+        metric_expiration: 180m <8>
+    service:
+      pipelines:
+        metrics:
+          exporters: [jaeger]
+----
+<1>  Exposes the Prometheus port from the collector pod and service. You can enable scraping of metrics by Prometheus by using the port name in `ServiceMonitor` or `PodMonitor`.
+<2> The endpoint on which the metrics will be exposed.
+<3> The server side TLS configuration. Defines paths to TLS certificates.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:434]:
```suggestion
<3> The server-side TLS configuration. Defines paths to TLS certificates.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 5]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+<4> If set, exports metrics under the provided value. No default.
+<5>  Key-value pair labels that are applied for every exported metric. No default.
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:436]:
```suggestion
<5> Key-value pair labels that are applied for every exported metric. No default.
```

**Comment by pavolloffay** [comment]
@max-cx PR updated

File: modules/distr-tracing-otel-config-collector.adoc [hunk 6]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+        const_labels: <5>
+          label1: value1
+        enable_open_metrics: true <6>
+        resource_to_telemetry_conversion: <7>
+          enabled: true
+        metric_expiration: 180m <8>
+    service:
+      pipelines:
+        metrics:
+          exporters: [jaeger]
```
Comment (IshwarKanse) [inline modules/distr-tracing-otel-config-collector.adoc:430]: exporters: [prometheus]

Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:430]:
```suggestion
          exporters: [prometheus]
```

Comment (pavolloffay) [reply modules/distr-tracing-otel-config-collector.adoc:430]: done

File: modules/distr-tracing-otel-config-collector.adoc [hunk 7]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+
+[id="prometheus-exporter_{context}"]
+==== Prometheus exporter
+
+The Prometheus exporter exports data using the Prometheus or OpenMetrics formats.
+
+* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
+* Supported signals: metrics
+
+.OpenTelemetry Collector custom resource with enabled Prometheus exporter
```
Comment (max-cx) [inline modules/distr-tracing-otel-config-collector.adoc:405]:
```suggestion
.OpenTelemetry Collector custom resource with the enabled Prometheus exporter
```

**Review by IshwarKanse — APPROVED** [review]

**Comment by max-cx** [comment]
Peer/language review is done.

**Comment by stevsmit** [comment]
@pavolloffay Two things:

1) I'm pushing this back to the peer review queue, as I don't see where a member of the OCP docs peer review squad did an initial review. This isn't a big deal; we'll still get it in today. 

2) Does this need cherry picked to 4.14? I see 4.11-4.13. I figure it should also go into 4.14 but I wanted to confirm.

File: modules/distr-tracing-otel-config-collector.adoc [hunk 8]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+        resource_to_telemetry_conversion: <7>
+          enabled: true
+        metric_expiration: 180m <8>
+    service:
+      pipelines:
+        metrics:
+          exporters: [prometheus]
+----
+<1> Exposes the Prometheus port from the collector pod and service. You can enable scraping of metrics by Prometheus by using the port name in `ServiceMonitor` or `PodMonitor`.
+<2> The endpoint on which the metrics will be exposed.
```
Comment (stevsmit) [inline modules/distr-tracing-otel-config-collector.adoc:433]: We don't use "will" in OCP docs: https://docs.engineering.redhat.com/display/CCSENABLE/Future+tense

```suggestion
<2>  The network endpoint where the metrics are exposed.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 9]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+        enable_open_metrics: true <6>
+        resource_to_telemetry_conversion: <7>
+          enabled: true
+        metric_expiration: 180m <8>
+    service:
+      pipelines:
+        metrics:
+          exporters: [prometheus]
+----
+<1> Exposes the Prometheus port from the collector pod and service. You can enable scraping of metrics by Prometheus by using the port name in `ServiceMonitor` or `PodMonitor`.
```
Comment (stevsmit) [inline modules/distr-tracing-otel-config-collector.adoc:432]: `ServiceMonitor` and `PodMonitor` need a noun after them. I think they're custom resources. 

```suggestion
<1> Exposes the Prometheus port from the collector pod and service. You can enable scraping of metrics by Prometheus by using the port name in `ServiceMonitor` or `PodMonitor`.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 10]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+<2> The endpoint on which the metrics will be exposed.
+<3> The server-side TLS configuration. Defines paths to TLS certificates.
+<4> If set, exports metrics under the provided value. No default.
+<5> Key-value pair labels that are applied for every exported metric. No default.
+<6> If `true`, metrics will be exported using the OpenMetrics format. Exemplars are only exported in the OpenMetrics format and only for histogram and monotonic sum metrics such as `counter`. Disabled by default.
```
Comment (stevsmit) [inline modules/distr-tracing-otel-config-collector.adoc:437]:
```suggestion
<6> If `true`, metrics are be exported using the OpenMetrics format. Exemplars are only exported in the OpenMetrics format and only for histogram and monotonic sum metrics such as `counter`. Disabled by default.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 11]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+<7> If `enabled` is `true`, all the resource attributes will be converted to metric labels by default. Disabled by default.
```
Comment (stevsmit) [inline modules/distr-tracing-otel-config-collector.adoc:438]:
```suggestion
<7> If `enabled` is `true`, all the resource attributes are converted to metric labels by default. Disabled by default.
```

**Comment by stevsmit** [comment]
@pavolloffay A few nits. Once these are fixed we can push to merge review.

File: modules/distr-tracing-otel-config-collector.adoc (hunk 9 above)
Comment (pavolloffay) [reply modules/distr-tracing-otel-config-collector.adoc:432]: done

**Comment by pavolloffay** [comment]
@stevsmit PR is updated

**Review by max-cx — APPROVED** [review]
LGTM.

File: modules/distr-tracing-otel-config-collector.adoc [hunk 12]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+        resource_to_telemetry_conversion: <7>
+          enabled: true
+        metric_expiration: 180m <8>
+    service:
+      pipelines:
+        metrics:
+          exporters: [prometheus]
+----
+<1> Exposes the Prometheus port from the collector pod and service. You can enable scraping of metrics by Prometheus by using the port name in `ServiceMonitor` or `PodMonitor` custom resource.
+<2> The network endpoint where the metrics are be exposed.
```
Comment (mburke5678) [inline modules/distr-tracing-otel-config-collector.adoc:433]:
```suggestion
<2> The network endpoint where the metrics are exposed.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 13]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+<3> The server-side TLS configuration. Defines paths to TLS certificates.
+<4> If set, exports metrics under the provided value. No default.
+<5> Key-value pair labels that are applied for every exported metric. No default.
+<6> If `true`, metrics are be exported using the OpenMetrics format. Exemplars are only exported in the OpenMetrics format and only for histogram and monotonic sum metrics such as `counter`. Disabled by default.
+<7> If `enabled` is `true`, all the resource attributes are be converted to metric labels by default. Disabled by default.
```
Comment (mburke5678) [inline modules/distr-tracing-otel-config-collector.adoc:438]:
```suggestion
<7> If `enabled` is `true`, all the resource attributes are converted to metric labels by default. Disabled by default.
```

File: modules/distr-tracing-otel-config-collector.adoc [hunk 14]
```diff
@@ -393,3 +393,48 @@ The Logging exporter prints data to the standard output.
...
+      pipelines:
+        metrics:
+          exporters: [prometheus]
+----
+<1> Exposes the Prometheus port from the collector pod and service. You can enable scraping of metrics by Prometheus by using the port name in `ServiceMonitor` or `PodMonitor` custom resource.
+<2> The network endpoint where the metrics are be exposed.
+<3> The server-side TLS configuration. Defines paths to TLS certificates.
+<4> If set, exports metrics under the provided value. No default.
+<5> Key-value pair labels that are applied for every exported metric. No default.
+<6> If `true`, metrics are be exported using the OpenMetrics format. Exemplars are only exported in the OpenMetrics format and only for histogram and monotonic sum metrics such as `counter`. Disabled by default.
```
Comment (mburke5678) [inline modules/distr-tracing-otel-config-collector.adoc:437]: Do we need to define what an exemplar is? I do not see it in the docs (other than vague references in the API docs)

Comment (pavolloffay) [reply modules/distr-tracing-otel-config-collector.adoc:437]: no, Exemplars are part of  OpenMetrics format, the customers that will be enabling this should know what it is.

**Comment by pavolloffay** [comment]
@mburke5678 PR is updated

**Review by IshwarKanse — APPROVED** [review]

**Comment by mburke5678** [comment]
@pavolloffay It seems that the `distr-tracing-otel-config-collector.adoc` file is not in [enterprise-4.11](https://github.com/openshift/openshift-docs/blob/enterprise-4.11/modules/distr-tracing-otel-config-collector.adoc) or [enterprise-4.12](https://github.com/openshift/openshift-docs/tree/enterprise-4.12/modules/distr-tracing-otel-config-collector.adoc).  I will cherrypick to 4.13 and 4.14. Please let me know if I am not correct.

**Comment by max-cx** [comment]
@mburke5678 and @pavolloffay, I believe the reason is the parent PR (https://github.com/openshift/openshift-docs/pull/62410) has not been cherry-picked.

**Comment by max-cx** [comment]
@mburke5678, please see
https://github.com/openshift/openshift-docs/pull/64716#issuecomment-1752966207
and
https://github.com/openshift/openshift-docs/pull/64716#issuecomment-1752966378
above.

**Pull request merged**